tst/EnergyPlus/unit/DataZoneEquipment.unit.cc

```diff
@@ -155,3 +160,269 @@ TEST_F(EnergyPlusFixture, DataZoneEquipment_TestCalcDesignSpecificationOutdoorAi
+    }
+  },
+  "Fan:ZoneExhaust": {
+    "Fan Zone Exhaust 1": {
+      "air_inlet_node_name": "Node 2",
+      "air_outlet_node_name": "Node 3",
+      "end_use_subcategory": "General",
+      "fan_total_efficiency": 0.6,
+      "pressure_rise": 0.0,
+      "system_availability_manager_coupling_mode": "Decoupled"
+    }
+  },
+  "ZoneHVAC:AirDistributionUnit": {
+    "ADU Air Terminal Single Duct Constant Volume No Reheat 1": {
+      "air_distribution_unit_outlet_node_name": "Node 8",
+      "air_terminal_name": "Air Terminal Single Duct Constant Volume No Reheat 1",
+      "air_terminal_object_type": "AirTerminal:SingleDuct:ConstantVolume:NoReheat"
+    },
+    "ADU Air Terminal Single Duct VAV Reheat 1": {
+      "air_distribution_unit_outlet_node_name": "Node 20",
+      "air_terminal_name": "Air Terminal Single Duct VAV Reheat 1",
+      "air_terminal_object_type": "AirTerminal:SingleDuct:VAV:Reheat"
+    }
+  },
+  "AirLoopHVAC:ZoneMixer": {
+    "Air Loop HVAC Zone Mixer 1": {
+      "nodes": [
+        {
+          "inlet_node_name": "Node 15"
+        }
+      ],
+      "outlet_node_name": "Node 7"
+    },
+    "Air Loop HVAC Zone Mixer 2": {
+      "nodes": [
+        {
+          "inlet_node_name": "Node 77"
+        }
+      ],
+      "outlet_node_name": "Node 19"
+    }
+  },
+  "AirLoopHVAC:ZoneSplitter": {
+    "Air Loop HVAC Zone Splitter 1": {
+      "inlet_node_name": "Node 6",
+      "nodes": [
+        {
+          "outlet_node_name": "Node 14"
+        }
+      ]
+    },
+    "Air Loop HVAC Zone Splitter 2": {
+      "inlet_node_name": "Node 18",
+      "nodes": [
+        {
+          "outlet_node_name": "Node 54"
+        }
+      ]
+    }
+  }
+}
+    )"_json;
+
+    state->dataGlobal->isEpJSON = true;
+    state->dataInputProcessing->inputProcessor->initializeMaps();
+
+    // GetZoneData and others use GetObjectItem with ipshortcuts so these need to be allocated
+    int MaxArgs = 0;
+    int MaxAlpha = 0;
+    int MaxNumeric = 0;
+    state->dataInputProcessing->inputProcessor->getMaxSchemaArgs(MaxArgs, MaxAlpha, MaxNumeric);
+
+    state->dataIPShortCut->cAlphaFieldNames.allocate(MaxAlpha);
+    state->dataIPShortCut->cAlphaArgs.allocate(MaxAlpha);
+    state->dataIPShortCut->lAlphaFieldBlanks.dimension(MaxAlpha, false);
+    state->dataIPShortCut->cNumericFieldNames.allocate(MaxNumeric);
+    state->dataIPShortCut->rNumericArgs.dimension(MaxNumeric, 0.0);
+    state->dataIPShortCut->lNumericFieldBlanks.dimension(MaxNumeric, false);
+
+    state->dataGlobal->NumOfTimeStepInHour = 1;    // must initialize this to get schedules initialized
+    state->dataGlobal->MinutesPerTimeStep = 60;    // must initialize this to get schedules initialized
+    ScheduleManager::ProcessScheduleInput(*state); // read schedules
+
+    bool ErrorsFound = false;
+    HeatBalanceManager::GetZoneData(*state, ErrorsFound);
+    ASSERT_FALSE(ErrorsFound);
+
+    DataZoneEquipment::GetZoneEquipmentData(*state);
```

Comment:
It throws here, because it thinks that are only 2 equipments on the ZoneHVAC:EquipmentList